federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -528,6 +532,12 @@ func (s *ServiceController) ensureClusterService(cachedService *cachedService, c
 	var err error
 	var needUpdate bool
 	for i := 0; i < clientRetryCount; i++ {
+		if key, err := cache.MetaNamespaceKeyFunc(service); err == nil {
+			if _, ok := s.serviceCache.get(key); !ok {
+				glog.Info(fmt.Sprintf("Service %v has been marked for deletion", key))
```

Comment:
This doesnt seem right. Its not removed from serviceCache until its actually deleted right?
When its marked for deletion, only DeletionTimestamp is set, but it still remains in serviceCache.